crypto/signature.go

```diff
@@ -30,74 +31,73 @@ func (e SignatureVerificationError) Error() string {
 // Internal functions
 // ------------------
 
-// newSignatureFailed creates a new SignatureVerificationError, type SIGNATURE_FAILED
+// newSignatureFailed creates a new SignatureVerificationError, type
+// SignatureFailed.
 func newSignatureFailed() SignatureVerificationError {
 	return SignatureVerificationError{
 		constants.SIGNATURE_FAILED,
 		"Invalid signature",
 	}
 }
 
-// newSignatureNotSigned creates a new SignatureVerificationError, type SIGNATURE_NOT_SIGNED
+// newSignatureNotSigned creates a new SignatureVerificationError, type
+// SignatureNotSigned.
 func newSignatureNotSigned() SignatureVerificationError {
 	return SignatureVerificationError{
 		constants.SIGNATURE_NOT_SIGNED,
 		"Missing signature",
 	}
 }
 
-// newSignatureNoVerifier creates a new SignatureVerificationError, type SIGNATURE_NO_VERIFIER
+// newSignatureNoVerifier creates a new SignatureVerificationError, type
+// SignatureNoVerifier.
 func newSignatureNoVerifier() SignatureVerificationError {
 	return SignatureVerificationError{
 		constants.SIGNATURE_NO_VERIFIER,
 		"No matching signature",
 	}
 }
 
-// processSignatureExpiration handles signature time verification manually, so we can add a margin to the
-// creationTime check.
+// processSignatureExpiration handles signature time verification manually, so
+// we can add a margin to the creationTime check.
 func processSignatureExpiration(md *openpgp.MessageDetails, verifyTime int64) {
-	if md.SignatureError == pgpErrors.ErrSignatureExpired {
-		if verifyTime > 0 {
-			created := md.Signature.CreationTime.Unix()
-			expires := int64(math.MaxInt64)
-			if md.Signature.SigLifetimeSecs != nil {
-				expires = int64(*md.Signature.SigLifetimeSecs) + created
-			}
-			if created-internal.CreationTimeOffset <= verifyTime && verifyTime <= expires {
-				md.SignatureError = nil
-			}
-		} else {
-			// verifyTime = 0: time check disabled, everything is okay
-			md.SignatureError = nil
-		}
+	if md.SignatureError != pgpErrors.ErrSignatureExpired {
+		return
+	}
+	if verifyTime <= 0 {
```

Comment:
I know that this is the correct refactor, but I think we should stick to checking `verifyTime == 0`